### PR TITLE
WIP: try kwant 1.4.2

### PR DESCRIPTION
**Pull request by akhmerov**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by akhmerov** [comment]
@conda-forge-admin please rerender

**Comment by akhmerov** [comment]
@conda-forge-admin please rerender

**Comment by basnijholt** [comment]
Windows is failing 😅

But it's just a test that is a bit outside of its tolerance:
```
________________________________ test_kwant_op ________________________________

    def test_kwant_op():
        """Check that the kwant.operator.Density gives the same result as the
        identity operator.
        """
        syst = make_chain()
        kwant_op = kwant.operator.Density(syst)
        spectrum_syst = make_spectrum(syst, p, operator=kwant_op, rng=1)
    
        ham = syst.hamiltonian_submatrix()
        identity = np.identity(dim)
        spectrum = make_spectrum(ham, p, operator=identity, rng=1)
    
        assert spectrum_syst.densities.shape[1] == ham.shape[0]
        # same algorithms are used so these arrays are equal up to TOL
        assert_allclose(np.sum(spectrum_syst.densities, axis=1),
>                       spectrum.densities)

..\_test_env\lib\site-packages\kwant\tests\test_kpm.py:422: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

arr1 = array([1.48245754e-01-1.13305112e-12j, 2.70321150e-02-3.30411886e-13j,
       1.07551259e-02-1.90250281e-13j, 1.971087....47758357e-14j, 9.97496981e-03-1.46812494e-14j,
       2.22703778e-02-8.73750670e-14j, 1.34796591e-01-2.95570643e-13j])
arr2 = array([1.48245754e-01, 2.70321150e-02, 1.07551259e-02, 1.97108730e-02,
       2.42624212e-02, 1.28219230e-02, 3.219773...3.51066462e-03, 9.79814214e-03, 2.05305631e-02,
       1.87794066e-02, 9.97496981e-03, 2.22703778e-02, 1.34796591e-01])

    def assert_allclose(arr1, arr2):
>       np.testing.assert_allclose(arr1, arr2, rtol=0., atol=TOL)
E       AssertionError: 
```

**Comment by akhmerov** [comment]
What worries me more in the failing test is not the tolerance, but that the dtypes are different.

**Comment by basnijholt** [comment]
Should we disable building on Windows for now?

**Comment by akhmerov** [comment]
Rather let's patch it. I'd like to investigate the type mismatch, but it doesn't look important.

**Comment by basnijholt** [comment]
I have patched the test.

**Comment by akhmerov** [comment]
Still TODO: wait until tagging, point the source to pypi.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,12 +7,12 @@ package:
 
 source:
   fn: {{ name }}-{{ version }}.tar.gz
-  url: https://downloads.kwant-project.org/kwant/kwant-{{ version }}.tar.gz
-  sha256: 8c0ccf341dfa74e1d69f1508968c4d4e9fb36f74685f101897df6a84ed7043df
+  url: https://antonakhmerov.org/misc/kwant-1.4.1.dev20+g40838439.tar.gz
```
Comment (jbweston) [inline recipe/meta.yaml]: Don't forget to change this back when the release is made

File: recipe/skip_test_kpm_kwant_op.patch [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+ import pytest
+@@ -404,6 +405,7 @@ def test_kwant_syst():
+     assert_allclose(spectrum_syst.densities, spectrum_dense.densities)
+
+
++@pytest.mark.skipif(sys.platform == 'win32', reason="fails on conda-forge")
+ def test_kwant_op():
+     """Check that the kwant.operator.Density gives the same result as the
+     identity operator.
```
Comment (jbweston) [inline recipe/skip_test_kpm_kwant_op.patch]: Shouldn't this problem be enough to block the release?

It's a bit strange that *only* this (seemingly simple?) test should fail.

File: recipe/meta.yaml (hunk 1 above)
Comment (akhmerov) [reply recipe/meta.yaml]: Of course. I'll switch this to PyPi for a source.

File: recipe/skip_test_kpm_kwant_op.patch (hunk 2 above)
Comment (akhmerov) [reply recipe/skip_test_kpm_kwant_op.patch]: The error was in the range of 1e-12, so we reckoned that it isn't a real failure.

**Comment by basnijholt** [comment]
@conda-forge-admin, please rerender

**Comment by akhmerov** [comment]
@basnijholt let's use pypi for canonical source, so that bots can pick it up from there.

**Comment by basnijholt** [comment]
This PR is superseded by #79 because some CI tasks refuse to update using `conda-smithy`.

**Pull request closed without merging**